client/lib/menu-data/menu-data.js

```diff
@@ -388,30 +407,76 @@ MenuData.prototype.saveMenu = function( menu, callback ) {
 
 	debug( 'saveMenu', menu );
 
-	this.emit( 'saving' );
-	wpcom
-	.undocumented()
-	.menusUpdate(
-		this.siteID,
-		menu.id,
-		this.interceptSaveForHomepageLink( menu ),
-		( error, data ) => {
-			if ( error ) {
-				this.emit( 'error', i18n.translate( 'There was a problem saving your menu.' ) );
-				debug( 'Error', error, data );
-				return;
-			}
+	const self = this;
+
+	function saveMenu() {
+		self.emit( 'saving' );
+		wpcom
+		.undocumented()
+		.menusUpdate(
+			self.siteID,
+			menu.id,
+			self.interceptSaveForHomepageLink( menu ),
+			( error, data ) => {
+				if ( error ) {
+					self.emit( 'error', i18n.translate( 'There was a problem saving your menu.' ) );
+					debug( 'Error', error, data );
+					return;
+				}
 
-			// The response will contain server-allocated
-			// IDs for newly created items
-			const parsedMenu = this.parseMenu( data.menu );
-			parsedMenu.lastSaveTime = Date.now();
-			this.replaceMenu( parsedMenu );
+				// The response will contain server-allocated
+				// IDs for newly created items
+				const parsedMenu = self.parseMenu( data.menu );
+				parsedMenu.lastSaveTime = Date.now();
+				self.replaceMenu( parsedMenu );
 
-			this.change( { reset: true } );
-			this.emit( 'saved' );
-			callback && callback( null, parsedMenu );
-		} );
+				self.change( { reset: true } );
+				self.emit( 'saved' );
+				callback && callback( null, parsedMenu );
+			} );
+	}
+
+	function nextNewPageCall( previousCall, menuItem ) {
+		return previousCall.then( () => {
+			return new Promise( ( resolve, reject ) => {
```

Comment:
Well, this is not possible, because Menu is serialized and saved "All at once".
These promises sequencially create new pages, but we cannot save "1 menu item".
(Mostly because menus are stored as JSON in DB)
Thats why it's called only after promises for page creation go through.